libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,432 @@
+        let request = self.request(forSearchScopeAndOrder: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a searchResultLayout
+    /// - Parameter objectList: Array of objects.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchResultLayout(_ objectList:[String], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchResultLayout: objectList.joined(separator: ","), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+     
+    /// Execute a Composite request
+    /// - Parameter requests: Array of RestRequewst
+    /// - Parameter referenceIds: String array of reference identifiers.
+    /// - Parameter allOrNone: Boolean to indicate whether to set allOrNone.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func composite(_ requests:[RestRequest], referenceIds: [String], allOrNone: Bool, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.compositeRequest(requests, refIds: referenceIds, allOrNone: allOrNone,apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a batch request
+    /// - Parameter requests: Array of RestRequewst
+    /// - Parameter haltOnError: Boolean to indicate whether to stop or continue when error occurs in any request.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func batch(_ requests:[RestRequest], haltOnError: Bool, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.batchRequest(requests, haltOnError: haltOnError, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute a SObjectTree request
+    /// - Parameter objectType: An object type.
+    /// - Parameter objectTrees: Array of SObjectTree
+    /// - Parameter completionBlock: The completion block to invoke.
+    func sObjectTree(_ objectType: String, objectTrees: [SObjectTree], _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSObjectTree: objectType, objectTrees: objectTrees, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Owned Files request
+    /// - Parameter userId: User identifier.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func ownedFiles(_ userId: String?, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
```

Comment:
I think `userId` should be required, not optional.

Reply:
@bhariharan its merely reflecting what the [obj-c api](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/ffa6ca5c31008ca41756eff2a357c45660bdd94d/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI%2BFiles.h#L42) does. Should we change the objc-api and impl? The obj-c api replaces userId with "ME" in the url path.